### PR TITLE
cleanup: resolve errorlint issues

### PR DESCRIPTION
**Pull request by yati1998**
This commit resolves errorlint issues
which checks for the code that will cause
problems with the error wrapping scheme.

Updates: #1586

Signed-off-by: Yati Padia <ypadia@redhat.com>

### CODE REVIEW

**Review by Madhu-1 — DISMISSED** [review]

File: internal/util/pidlimit.go [hunk 1]
```diff
@@ -78,7 +79,7 @@ func GetPIDLimit() (int, error) {
 	defer f.Close() // #nosec: error on close is not critical here
 
 	maxPidsStr, err := bufio.NewReader(f).ReadString('\n')
-	if err != nil && err != io.EOF {
+	if !errors.Is(err, nil) && !errors.Is(err, io.EOF) {
```
Comment (nixpanic) [inline internal/util/pidlimit.go]: `!errors.Is(err, nil)` should just be `err != nil` :wink:

Checking `io.EOF` is an exception, I think. It does not need `errors.Is`. Did the linter really complain about this?

**Review by nixpanic — CHANGES_REQUESTED** [review]
just a minor change needed, looks good otherwise

File: internal/util/pidlimit.go (hunk 1 above)
Comment (yati1998) [reply internal/util/pidlimit.go]: `internal/util/pidlimit.go:82:29: comparing with != will fail on wrapped errors. Use errors.Is to check for a specific error (errorlint)
	if !errors.Is(err, nil) && err != io.EOF {`

Comment (nixpanic) [reply internal/util/pidlimit.go]: hmm, ok. But still need to change back to `err != nil`, as `nil` is not a wrapper error.

**Review by nixpanic — CHANGES_REQUESTED** [review]
minor change needed. Also update the commit + PR description with `Updates: #1586` (now missing the `#` and therefore not having the links generated)

File: internal/util/pidlimit.go (hunk 1 above)
Comment (Madhu-1) [reply internal/util/pidlimit.go]:
```suggestion
if err != nil{
    if  !errors.Is(err, io.EOF) {
      return 0, err
    }
}

Comment (yati1998) [reply internal/util/pidlimit.go]: `if err != nil && !errors.Is(err, io.EOF) ` was working. Hence changed to that only.

**Review by Madhu-1 — DISMISSED** [review]

**Review by nixpanic — DISMISSED** [review]

File: e2e/utils.go [hunk 2]
```diff
@@ -1091,7 +1091,7 @@ func validateController(f *framework.Framework, pvcPath, appPath, scPath string)
 	pv.ResourceVersion = ""
 	err = createPVCAndPV(f.ClientSet, pvc, pv)
 	if err != nil {
-		e2elog.Failf("failed to create PVC or PV with error %v", err)
+		e2elog.Failf("failed to create PVC or PV: %v", err)
```
Comment (pkalever) [inline e2e/utils.go:1094]: `%w` here?

**Review by pkalever — CHANGES_REQUESTED** [review]

File: e2e/utils.go (hunk 2 above)
Comment (nixpanic) [reply e2e/utils.go:1094]: No, `%w` is for returning a wrapped error, not for logging an error.

**Comment by nixpanic** [comment]
> /retest ci/centos/mini-e2e/k8s-1.20

`NodePublishVolume` for CephFS PV pvc-f9250fb8-767c-44c9-bbed-130c6d2c4c56 with request-id 31 never finished ([logs](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.20/detail/mini-e2e_k8s-1.20/1647/pipeline)):
```
I0714 14:28:46.139903       1 utils.go:171] ID: 31 Req-ID: 0001-0024-60ab47e6-e3ab-479e-a924-698281ed1b3b-0000000000000001-c9ebee2e-e4af-11eb-9b57-4e25d0b69dca GRPC call: /csi.v1.Node/NodePublishVolume
I0714 14:28:46.140105       1 utils.go:175] ID: 31 Req-ID: 0001-0024-60ab47e6-e3ab-479e-a924-698281ed1b3b-0000000000000001-c9ebee2e-e4af-11eb-9b57-4e25d0b69dca GRPC request: {"staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-f9250fb8-767c-44c9-bbed-130c6d2c4c56/globalmount","target_path":"/var/lib/kubelet/pods/7994b9c8-481c-48be-9c62-f871d2d4aa0c/volumes/kubernetes.io~csi/pvc-f9250fb8-767c-44c9-bbed-130c6d2c4c56/mount","volume_capability":{"AccessType":{"Mount":{"mount_flags":["debug"]}},"access_mode":{"mode":5}},"volume_context":{"clusterID":"60ab47e6-e3ab-479e-a924-698281ed1b3b","csi.storage.k8s.io/pv/name":"pvc-f9250fb8-767c-44c9-bbed-130c6d2c4c56","csi.storage.k8s.io/pvc/name":"csi-cephfs-pvc","csi.storage.k8s.io/pvc/namespace":"cephfs-8081","fsName":"myfs","fuseMountOptions":"debug","mounter":"fuse","pool":"myfs-data0","storage.kubernetes.io/csiProvisionerIdentity":"1626272803856-8081-cephfs.csi.ceph.com","subvolumeName":"csi-vol-c9ebee2e-e4af-11eb-9b57-4e25d0b69dca","subvolumePath":"/volumes/e2e/csi-vol-c9ebee2e-e4af-11eb-9b57-4e25d0b69dca/bc645063-5708-4472-8b39-ccdfb94e7603"},"volume_id":"0001-0024-60ab47e6-e3ab-479e-a924-698281ed1b3b-0000000000000001-c9ebee2e-e4af-11eb-9b57-4e25d0b69dca"}
I0714 14:28:46.151946       1 mount_linux.go:173] Cannot run systemd-run, assuming non-systemd OS
I0714 14:28:46.151978       1 mount_linux.go:174] systemd-run failed with: exit status 1
I0714 14:28:46.151993       1 mount_linux.go:175] systemd-run output: System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to create bus connection: Host is down
I0714 14:29:28.249554       1 utils.go:171] ID: 32 GRPC call: /csi.v1.Identity/Probe
I0714 14:29:28.249639       1 utils.go:175] ID: 32 GRPC request: {}
I0714 14:29:28.249677       1 utils.go:182] ID: 32 GRPC response: {}
I0714 14:30:28.250114       1 utils.go:171] ID: 33 GRPC call: /csi.v1.Identity/Probe
I0714 14:30:28.250197       1 utils.go:175] ID: 33 GRPC request: {}
I0714 14:30:28.250231       1 utils.go:182] ID: 33 GRPC response: {}
I0714 14:30:46.659598       1 utils.go:171] ID: 34 GRPC call: /csi.v1.Node/NodeGetCapabilities
I0714 14:30:46.659688       1 utils.go:175] ID: 34 GRPC request: {}
I0714 14:30:46.659817       1 utils.go:182] ID: 34 GRPC response: {"capabilities":[{"Type":{"Rpc":{"type":1}}},{"Type":{"Rpc":{"type":2}}}]}
I0714 14:30:46.664384       1 utils.go:171] ID: 35 Req-ID: 0001-0024-60ab47e6-e3ab-479e-a924-698281ed1b3b-0000000000000001-c9ebee2e-e4af-11eb-9b57-4e25d0b69dca GRPC call: /csi.v1.Node/NodePublishVolume
I0714 14:30:46.664594       1 utils.go:175] ID: 35 Req-ID: 0001-0024-60ab47e6-e3ab-479e-a924-698281ed1b3b-0000000000000001-c9ebee2e-e4af-11eb-9b57-4e25d0b69dca GRPC request: {"staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-f9250fb8-767c-44c9-bbed-130c6d2c4c56/globalmount","target_path":"/var/lib/kubelet/pods/7994b9c8-481c-48be-9c62-f871d2d4aa0c/volumes/kubernetes.io~csi/pvc-f9250fb8-767c-44c9-bbed-130c6d2c4c56/mount","volume_capability":{"AccessType":{"Mount":{"mount_flags":["debug"]}},"access_mode":{"mode":5}},"volume_context":{"clusterID":"60ab47e6-e3ab-479e-a924-698281ed1b3b","csi.storage.k8s.io/pv/name":"pvc-f9250fb8-767c-44c9-bbed-130c6d2c4c56","csi.storage.k8s.io/pvc/name":"csi-cephfs-pvc","csi.storage.k8s.io/pvc/namespace":"cephfs-8081","fsName":"myfs","fuseMountOptions":"debug","mounter":"fuse","pool":"myfs-data0","storage.kubernetes.io/csiProvisionerIdentity":"1626272803856-8081-cephfs.csi.ceph.com","subvolumeName":"csi-vol-c9ebee2e-e4af-11eb-9b57-4e25d0b69dca","subvolumePath":"/volumes/e2e/csi-vol-c9ebee2e-e4af-11eb-9b57-4e25d0b69dca/bc645063-5708-4472-8b39-ccdfb94e7603"},"volume_id":"0001-0024-60ab47e6-e3ab-479e-a924-698281ed1b3b-0000000000000001-c9ebee2e-e4af-11eb-9b57-4e25d0b69dca"}
E0714 14:30:46.664632       1 nodeserver.go:224] ID: 35 Req-ID: 0001-0024-60ab47e6-e3ab-479e-a924-698281ed1b3b-0000000000000001-c9ebee2e-e4af-11eb-9b57-4e25d0b69dca an operation with the given Volume ID 0001-0024-60ab47e6-e3ab-479e-a924-698281ed1b3b-0000000000000001-c9ebee2e-e4af-11eb-9b57-4e25d0b69dca already exists
E0714 14:30:46.664656       1 utils.go:180] ID: 35 Req-ID: 0001-0024-60ab47e6-e3ab-479e-a924-698281ed1b3b-0000000000000001-c9ebee2e-e4af-11eb-9b57-4e25d0b69dca GRPC error: rpc error: code = Aborted desc = an operation with the given Volume ID 0001-0024-60ab47e6-e3ab-479e-a924-698281ed1b3b-0000000000000001-c9ebee2e-e4af-11eb-9b57-4e25d0b69dca already exists
```

File: e2e/utils.go (hunk 2 above)
Comment (pkalever) [reply e2e/utils.go:1094]: Yeah, sorry. Didn't notice that this was with logging.

**Review by nixpanic — APPROVED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Comment by nixpanic** [comment]
> /retest ci/centos/mini-e2e-helm/k8s-1.21

Failed due to #2264

**Comment by nixpanic** [comment]
> /retest ci/centos/mini-e2e-helm/k8s-1.19

Failed with some etcdserver timeouts ([logs](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/mini-e2e-helm_k8s-1.19/detail/mini-e2e-helm_k8s-1.19/2090/pipeline)):
```
Jul 19 08:06:17.613: INFO: waiting for PVC rbd-pvc (0 seconds elapsed)
Jul 19 08:06:19.623: INFO: waiting for PVC rbd-pvc (2 seconds elapsed)
Jul 19 08:06:21.621: INFO: waiting for PVC rbd-pvc (4 seconds elapsed)
Jul 19 08:06:23.622: INFO: waiting for PVC rbd-pvc (6 seconds elapsed)
Jul 19 08:06:33.622: INFO: waiting for PVC rbd-pvc (16 seconds elapsed)
Jul 19 08:06:45.928: INFO: Error getting pvc "rbd-pvc" in namespace "rbd-1318": etcdserver: request timed out
Jul 19 08:06:47.623: INFO: waiting for PVC rbd-pvc (30 seconds elapsed)
Jul 19 08:06:56.324: INFO: Error getting pvc "rbd-pvc" in namespace "rbd-1318": etcdserver: request timed out
Jul 19 08:06:57.622: INFO: waiting for PVC rbd-pvc (40 seconds elapsed)
Jul 19 08:07:09.622: INFO: waiting for PVC rbd-pvc (52 seconds elapsed)
Jul 19 08:07:21.622: INFO: waiting for PVC rbd-pvc (64 seconds elapsed)
Jul 19 08:07:25.622: INFO: waiting for PVC rbd-pvc (68 seconds elapsed)
Jul 19 08:07:38.417: INFO: Error getting pvc "rbd-pvc" in namespace "rbd-1318": etcdserver: request timed out
Jul 19 08:07:39.622: INFO: waiting for PVC rbd-pvc (82 seconds elapsed)
Jul 19 08:07:47.622: INFO: waiting for PVC rbd-pvc (90 seconds elapsed)
Jul 19 08:07:59.624: INFO: waiting for PVC rbd-pvc (102 seconds elapsed)
Jul 19 08:07:59.634: INFO: Error getting pvc "rbd-pvc" in namespace "rbd-1318": Get "https://192.168.39.222:8443/api/v1/namespaces/rbd-1318/persistentvolumeclaims/rbd-pvc": dial tcp 192.168.39.222:8443: connect: connection refused
Jul 19 08:07:59.634: FAIL: failed to create PVC with error failed to get pvc: Get "https://192.168.39.222:8443/api/v1/namespaces/rbd-1318/persistentvolumeclaims/rbd-pvc": dial tcp 192.168.39.222:8443: connect: connection refused
```

**Comment by nixpanic** [comment]
> /retest ci/centos/mini-e2e-helm/k8s-1.20

Deploying minikube+rook timed out, network slowness in the CI infra? ([logs](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/mini-e2e-helm_k8s-1.20/detail/mini-e2e-helm_k8s-1.20/1701/pipeline/74))

**Pull request merged**